### ISSUE TITLE
Way to run tests on ci

### ISSUE THREAD

**Issue by gintama91** [gintama91/wry_ruby#3]
refer  [this](https://github.com/gintama91/wry_ruby/pull/2#issuecomment-1614218242) 
and also think about do we really need testing this on ci ?

**Comment by gintama91** [comment]
https://github.com/electron/electron/blob/main/docs/tutorial/testing-on-headless-ci.md

**Comment by gintama91** [comment]
this is using docker  https://github.com/rlespinasse/docker-drawio-desktop-headless

**Comment by gintama91** [comment]
hm we prolly have to switch to RSpec or cucmber?

**Comment by noahgibbs** [comment]
If Wry is going to be a primary display service, we would really really need CI testing. If it's just a prototype, we don't necessarily need it.

**Comment by noahgibbs** [comment]
Xvfb could definitely work, and isn't a bad idea in any case - we could take screenshots on test runs, for instance. Here's an "install xvfb on GHActions" step: https://github.com/coactions/setup-xvfb

**Comment by noahgibbs** [comment]
Docker is possible but really heavy-weight. It would take time to start and stop, and use a lot of memory. Possible, but it would be good to avoid it if we can.

**Comment by noahgibbs** [comment]
I don't think xvfb or docker should require changing our test harness, though. We might just have to add some kind of config. But minitest should be exactly as compatible (or incompatible) as RSpec with xvfb or docker, with a given configuration.

**Comment by gintama91** [comment]
ok so i have tried xvfb its working fine but as of now we have to implement something tht will close the window.(as if we dont close tests task is keep running infinitely) we cant run it in diff thread so until tht this has to be on hold ig

https://github.com/gintama91/wry_ruby/actions/runs/5436821030/jobs/9886937099

in `ci-dev` branch

**Comment by noahgibbs** [comment]
Yeah, we have a whole thing for Webview where we shut it down on a timer. You can see the code in test_helper.rb. But yeah, you'd have to do something similar for wry.

**Comment by gintama91** [comment]
hm if u r refering to doing something like
```
require "test_helper"
require "timeout"

class TestWryRuby < Minitest::Test
  def test_hello_wry
    timeout_seconds = 1
    assert_timeout(timeout_seconds) do
      new_window "Hello Wry", 100, 400, true # title, width, height, resizable
    end
  end

  private

  def assert_timeout(timeout_seconds, &block)
    Timeout.timeout(timeout_seconds, &block)
  rescue Timeout::Error
    flunk "Test timed out after #{timeout_seconds} seconds."
  end
end

```

it didn't worked

**Comment by gintama91** [comment]
this is possible from rust side

**Comment by gintama91** [comment]
ah ok idk why its not working on ci  it was workin on local.

**Comment by noahgibbs** [comment]
Yeah, something basically like that. Eventually you'll probably want something a lot like ControlInterface/ControlInterfaceTest for Webview. But you can start with something more like the timeout above, yeah.
